src/DssSpell.sol

```diff
@@ -19,132 +19,454 @@ pragma solidity 0.6.12;
-    address constant SF_001_WALLET         = 0xf737C76D2B358619f7ef696cf3F94548fEcec379;
-    address constant RWF_001_WALLET        = 0x96d7b01Cc25B141520C717fa369844d34FF116ec;
-    address constant SF_001_VEST_01        = 0xBC7fd5AA2016C3e2C8F0dBf4e919485C6BBb59e2;
-    address constant SF_001_VEST_02        = 0xCC81578d163A04ea8d2EaE6904d0C8E61A84E1Bb;
+    address constant NEW_MCD_ESM  = 0x09e05fF6142F2f9de8B6B65855A1d56B6cfE4c58;
+    bytes32 constant MCD_ESM_NAME = "MCD_ESM";
 
+    address constant FLIP_FLOP_FLAP_WALLET  = 0x688d508f3a6B0a377e266405A1583B3316f9A2B3;
+    address constant FEEDBLACK_LOOPS_WALLET = 0x80882f2A36d49fC46C3c654F7f9cB9a2Bf0423e1;
+    address constant SCHUPPI_WALLET         = 0x89C5d54C979f682F40b73a9FC39F338C88B434c6;
+    address constant MAKERMAN_WALLET        = 0x9AC6A6B24bCd789Fa59A175c0514f33255e1e6D0;
+    address constant MONETSUPPLY_WALLET     = 0x4Bd73eeE3d0568Bb7C52DFCad7AD5d47Fff5E2CF;
+    address constant ACRE_INVEST_WALLET     = 0x5b9C98e8A3D9Db6cd4B4B4C1F92D0A551D06F00D;
+    address constant JUSTIN_CASE_WALLET     = 0xE070c2dCfcf6C6409202A8a210f71D51dbAe9473;
+    address constant GFX_LABS_WALLET        = 0xa6e8772af29b29B9202a073f8E36f447689BEef6;
 
-    uint256 constant APR_01_2021 = 1617235200;
-    uint256 constant SEP_01_2021 = 1630454400;
-    uint256 constant FEB_01_2022 = 1643673600;
-    uint256 constant DEC_31_2022 = 1672444800;
-    uint256 constant JAN_31_2023 = 1675123200;
-    uint256 constant JUL_31_2023 = 1690761600;
+    VestAbstract immutable VEST             = VestAbstract(DssExecLib.getChangelogAddress("MCD_VEST_DAI"));
+    address constant SF_001_WALLET          = 0xf737C76D2B358619f7ef696cf3F94548fEcec379;
+    address constant SNE_001_WALLET         = 0x6D348f18c88D45243705D4fdEeB6538c6a9191F1;
 
+    uint256 constant MAR_01_2022            = 1646092800;
+    uint256 constant JUL_31_2022            = 1659225600;
 
     // Math
     uint256 constant MILLION = 10**6;
     uint256 constant WAD = 10**18;
+    uint256 constant RAY = 10**27;
 
-    // Turn office hours off
-    function officeHours() public override returns (bool) {
-        return false;
+    function sub(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        require((z = x - y) <= x, "sub-underflow");
     }
 
     function actions() public override {
 
         // Includes changes from the DssSpellCollateralOnboardingAction
         // onboardNewCollaterals();
 
-        // Revoking Content Production Budget (MKT-001)
-        // https://mips.makerdao.com/mips/details/MIP40c3SP49
-        DssVestLike(MCD_VEST_DAI).yank(20);
+        address MCD_VAT      = DssExecLib.getChangelogAddress("MCD_VAT");
+        address MCD_VOW      = DssExecLib.getChangelogAddress("MCD_VOW");
+        address MCD_JOIN_DAI = DssExecLib.getChangelogAddress("MCD_JOIN_DAI");
+        address MCD_VEST_DAI = DssExecLib.getChangelogAddress("MCD_VEST_DAI");
```

Comment:
This is an unused variable as there is another `VEST` one.
Btw we should put all of them in the immutable part to keep the consistency.

Reply:
Done both.